### DISCUSSION TITLE
Add a scripts folder to repo or scripts wiki section

### DISCUSSION THREAD

**Discussion by RayZ0rr** [i3/i3#4815] [Ideas]
Request to add a `scripts` folder to repo which will hold some example scripts that users may use directly or use as reference to extend functionality of i3

Can also add a wiki section to mention about scripts which also has the added benefit of having community-contributed scripts.

My small script contribution : 
```bash
#!/usr/bin/env bash

# https://github.com/RayZ0rr
#
# A simple script for workspace switching within a range
# specified by the variable $num_limit below.
#
# === Dependencies ===
# 1. jq (https://github.com/stedolan/jq)
#
# === Usage ===
#
# Put this script somewhere, say (~/.config/i3/scripts/workspace_changer.sh).
# Make sure it is executable:
#
#     chmod +x ~/.config/i3/scripts/workspace_change.sh
#
# Use as (use your desired key instead of 'n' and 'm'):
# bindsym $mod+n exec --no-startup-id ~/.config/i3/scripts/workspace_change.sh prev
# bindsym $mod+m exec --no-startup-id ~/.config/i3/scripts/workspace_change.sh next


num_current_ws=$(i3-msg -t get_workspaces | jq '.[] | select(.focused).num')

num_limit=10

num_minus=$((${num_current_ws} - 1))
num_plus=$((${num_current_ws} + 1))

case "${num_current_ws}" in
  "10")
    num_plus=1 ;;
  "1")
    num_minus=10 ;;
esac

if [[ "$@" = "next" ]] ; then
  i3-msg workspace number ${num_plus} &> /dev/null
  exit 0
elif [[ "$@" = "prev" ]] ; then
  i3-msg workspace number ${num_minus} &> /dev/null
  exit 0
else
  echo "invalid argument. Use 'next' or 'prev'"
  exit 1
fi

```

Also, relevant this [repo](https://github.com/budlabs/i3ass)

**Comment by Airblader** [comment]
This is the same as #4509, isn't it?

**Reply by RayZ0rr** [reply]
It is the same as one of the things that I mentioned in my post. **A mention about community resources** either on a wiki page or in a thread in this discussion. It can also be announced in the subreddit to make more users aware. As it's a community resource, then obviously it comes with a **use at your own risk** disclaimer or that can be explicitly put there.

What I primarily intended to mention was an **example scripts** folder in i3 itself which will show/provide some popular and useful functionalities to the user. Although functional, they will be there as references to users. An example is my script above. But other things I would like this folder to have : 
1) Save a workspace layout
2) Load a layout ( can be made into one?)

Something like these, but after the experts review it: 
save_layout.sh (WIP)
```bash
#!/usr/bin/env bash

! [[ -d "${HOME}/.config/i3/layouts" ]] && mkdir -p "${HOME}/.config/i3/layouts"

FI=`zenity --entry --title="Save the current Layout" --text="Enter name of new Layout:" --entry-text "NewLayout"`
i3-save-tree > ${HOME}/.config/i3/layouts/"$FI"  # Automate further processing. Add jq as dependency

# If you want to use dmenu instead of zenity
# FI=`dmenu -p "Enter the name of the new layout" `
# i3-save-tree > ${HOME}/.config/i3/layouts/$FI
```

**Reply by Airblader** [reply]
> What I primarily intended to mention was an example scripts folder in i3 itself

We have discussed this in the past and don't want to do it. The notion of a "use at your own risk" disclaimer is nice, but pointless; _we_ still need to maintain it. Having an externalized resource for this, be it here on Discussions or elsewhere, allows the community to manage this independently. But it does require for someone to step up, and it's much easier for people to demand something than to actually take on the responsibility of maintaining it.

**Reply by Airblader** [reply]
Addendum: if someone were to create such a place and demonstrate over time that they are committed to maintaining it, we can always talk about featuring it more prominently.

**Reply by RayZ0rr** [reply]
> We have discussed this in the past and don't want to do it. The notion of a "use at your own risk" disclaimer is nice, but pointless; _we_ still need to maintain it.

I am sorry for the lack of clarification but the example folder in i3 would only have scripts like the ones I posted in this thread (`workspace_change.sh` and fixed `save_layout.sh`). They won't be maintained after that. Many software comes with some examples folder in them that consists of scripts to utilize their software. Some of them work, some of them don't. Well, it's only examples.
> Having an externalized resource for this, be it here on Discussions or elsewhere, allows the community to manage this independently.

This is true and what is being discussed on #4509. I have pinged the author of that thread to see what are the updates on it.

> Addendum: if someone were to create such a place and demonstrate over time that they are committed to maintaining it, we can always talk about featuring it more prominently.

If it is mentioned in the official repo, then it's good enough I believe. I will see if it's possible to bring up a community resources repo like that.

**Reply by Airblader** [reply]
> They won't be maintained after that.

This is exactly why we don't want to add such a folder. Scripts that are not maintained are virtually useless, and the expectation to any resource part of i3 itself will (and should) be that it works with that version of i3.

**Reply by RayZ0rr** [reply]
Yes, I understand your point. That's why the scripts I posted here are very basic and simple. They will work as long as `i3-save-tree` and `i3-msg -t get_workspaces` doesn't change (which I believe is a very long time). As I said if an external repo comes up, I will raise these requests there.
